setup.py

```diff
@@ -61,6 +61,7 @@
         'six>=1.10.0',
         'Sphinx>=1.5.1',
         'Jinja2>=2.8',
+        'wrapt>=1.10.8',
```

Comment:
Good point. I decided to add `wrapt` as a dependency because it strongly [focuses on **correctness** when implementing decorators](http://wrapt.readthedocs.io/en/latest/):

> The wrapt module focuses very much on **correctness**. It therefore goes way beyond existing mechanisms such as `functools.wraps()` to ensure that decorators preserve introspectability, signatures, type checking abilities etc. The decorators that can be constructed using this module will work in far more scenarios than typical decorators and provide more predictable and consistent behaviour.

And because it has [very good documentation on how to create a complex decorator with arguments and optional arguments](http://wrapt.readthedocs.io/en/latest/decorators.html).